src/main/java/org/cryptomator/common/vaults/VaultListManager.java

```diff
@@ -134,29 +151,70 @@ public static VaultState.Value redetermineVaultState(Vault vault) {
 		VaultState state = vault.stateProperty();
 		VaultState.Value previousState = state.getValue();
 		return switch (previousState) {
-			case LOCKED, NEEDS_MIGRATION, MISSING -> {
+			case LOCKED, NEEDS_MIGRATION, MISSING, VAULT_CONFIG_MISSING, MASTERKEY_MISSING -> {
 				try {
-					var determinedState = determineVaultState(vault.getPath());
+					var determinedState = determineVaultState(vault.getPath(), vault.getVaultSettings());
+					if (determinedState == MASTERKEY_MISSING) {
+						var vaultScheme = vault.getVaultConfigCache().getUnchecked().getKeyId().getScheme();
+						if (KeyLoadingStrategy.isHubVault(vaultScheme)) {
+							determinedState = LOCKED;
+						}
+					}
 					if (determinedState == LOCKED) {
 						vault.getVaultConfigCache().reloadConfig();
 					}
 					state.set(determinedState);
 					yield determinedState;
 				} catch (IOException e) {
-					LOG.warn("Failed to determine vault state for " + vault.getPath(), e);
+					LOG.warn("Failed to determine vault state for {}", vault.getPath(), e);
 					state.set(ERROR);
 					vault.setLastKnownException(e);
 					yield ERROR;
 				}
 			}
-			case ERROR, UNLOCKED, PROCESSING -> previousState;
+			case ERROR -> {
+				try {
+					var determinedState = determineVaultState(vault.getPath(), vault.getVaultSettings());
+					state.set(determinedState);
+					yield determinedState;
+				} catch (IOException e) {
+					LOG.warn("Failed to redetermine vault state for {}", vault.getPath(), e);
+					vault.setLastKnownException(e);
+					yield ERROR;
+				}
+			}
+			case UNLOCKED, PROCESSING -> previousState;
 		};
 	}
 
-	private static VaultState.Value determineVaultState(Path pathToVault) throws IOException {
-		if (!Files.exists(pathToVault)) {
+	private static VaultState.Value determineVaultState(Path pathToVault, VaultSettings vaultSettings) throws IOException {
+		Path pathToVaultConfig = pathToVault.resolve("vault.cryptomator");
+		Path pathToMasterkey = pathToVault.resolve("masterkey.cryptomator");
+
+		if (Files.notExists(pathToVault)) {
```

Comment:
use Files.exists instead. From the [javadoc](https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/nio/file/Files.html#exists(java.nio.file.Path,java.nio.file.LinkOption...):
> [returns] true if the file exists; false if the file does not exist or its existence cannot be determined.